### PR TITLE
chore(storage-browser): refactor list types, navigation, bucket parsing

### PR DESCRIPTION
**Pull request by calebpollman**
#### Description of changes
Refactor list types, navigation, and location value parsing

#### Checklist

- [ ] Have read the [Pull Request Guidelines](https://github.com/aws-amplify/amplify-ui/blob/main/CONTRIBUTING.md)
- [ ] PR description included
- [ ] `yarn test` passes and tests are updated/added
- [ ] PR title and commit messages follow [conventional commit](https://www.conventionalcommits.org/en/v1.0.0/#summary) syntax
- [ ] _If this change should result in a version bump_, [changeset added](https://github.com/changesets/changesets/blob/main/docs/adding-a-changeset.md) (This can be done after creating the PR.) This does not apply to changes made to `docs`, `e2e`, `examples`, or other private packages.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by calebpollman** [comment]
CI failed due to target branch using an older version of the `aws-amplify@storage-manager` tag then what I have locally. Will refresh the version in a revision but failure should not impact reviewing

File: packages/react-storage/src/components/StorageBrowser/context/controls/Navigate/__tests__/Navigate.spec.tsx [hunk 1]
```diff
@@ -1,159 +1,106 @@
...
     };
 
     const newState = navigateReducer(state, action);
 
-    expect(newState).toEqual(expectedState);
+    expect(newState).toStrictEqual(expectedState);
   });
 
-  it('should handle ENTER_FOLDER with undefined history', () => {
+  it('handles a NAVIGATE action to a new location', () => {
```
Comment (timngyn) [inline packages/react-storage/src/components/StorageBrowser/context/controls/Navigate/__tests__/Navigate.spec.tsx:68]: not blocking, but I think having another test navigating back from a prefix could be helpful too?


```jsx
  it('handles a NAVIGATE action to an old location', () => {
    const state: NavigateState = {
      history: [location.bucket, locationTwo.prefix, 'public/folder1/', 'public/folder1/nestedFolder/'],
      location: locationAccess,
    };
    
    const action: NavigateAction = {
      type: 'NAVIGATE',
      prefix: locationTwo.prefix,
    };
    
    const expectedState: NavigateState = {
      history: [location.bucket, locationTwo.prefix],
      location: locationAccess,
    };

    const newState = navigateReducer(state, action);

    expect(newState).toStrictEqual(expectedState);
  }
```

File: packages/react-storage/src/components/StorageBrowser/Views/LocationDetailView/LocationDetailView.tsx [hunk 2]
```diff
@@ -7,74 +7,69 @@ import { LocationDetailViewControls } from './Controls';
...
-  const hasItems = !!data.items.length;
+    handleListLocationItems({
+      prefix: prefix === bucket ? '' : prefix,
+      config: {
+        bucket,
+        credentialsProvider: async () =>
+          await getLocationCredentials({ permission, scope }),
+        region,
+      },
+      options: { refresh: true },
```
Comment (timngyn) [inline packages/react-storage/src/components/StorageBrowser/Views/LocationDetailView/LocationDetailView.tsx:45]: do we still need the pageSize of this call to be 1000?

Comment (calebpollman) [reply packages/react-storage/src/components/StorageBrowser/Views/LocationDetailView/LocationDetailView.tsx:45]: Yes, good catch. Got lost in a rebase

File: packages/react-storage/src/components/StorageBrowser/adapters/createAmplifyAuthAdapter.ts [hunk 3]
```diff
@@ -22,6 +37,11 @@ export const createAmplifyAuthAdapter = (): StorageBrowserAuthAdapter => {
     return Promise.resolve({
       locations: [
         { type: 'BUCKET', permission: 'READWRITE', scope: `s3://${bucket}/*` },
+        {
+          type: 'PREFIX',
+          permission: 'READWRITE',
+          scope: `s3://${bucket}/public/*`,
+        },
```
Comment (timngyn) [inline packages/react-storage/src/components/StorageBrowser/adapters/createAmplifyAuthAdapter.ts:44]: Is this a temporary work around for the `default-auth` path or something like that?

**Review by timngyn — APPROVED** [review]
LGTM!

**Review by timngyn — APPROVED** [review]

**Pull request merged**